src/pages/workspace/WorkspacesListRow.tsx

```diff
@@ -0,0 +1,168 @@
+
+const workspaceTypeIcon = (workspaceType: WorkspacesListRowProps['workspaceType']): React.FC<SvgProps> => {
+    switch (workspaceType) {
+        case CONST.POLICY.TYPE.FREE:
+            return Illustrations.BigRocket;
+        case CONST.POLICY.TYPE.CORPORATE:
+            return Illustrations.HotDogStand;
+        case CONST.POLICY.TYPE.TEAM:
+            return Illustrations.SmallRocket;
+        default:
+            throw new Error(`Don't know which icon to serve for workspace type`);
+    }
+};
+
+function WorkspacesListRow({title, workspaceIcon, fallbackWorkspaceIcon, ownerAccountID, workspaceType, currentUserPersonalDetails, isWide}: WorkspacesListRowProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const ownerDetails = PersonalDetailsUtils.getPersonalDetailsByIDs([ownerAccountID], currentUserPersonalDetails.accountID)[0];
+
+    const userFriendlyWorkspaceType = useMemo(() => {
+        switch (workspaceType) {
+            case CONST.POLICY.TYPE.FREE:
+                return translate('workspace.type.free');
+            case CONST.POLICY.TYPE.CORPORATE:
+                return translate('workspace.type.control');
+            case CONST.POLICY.TYPE.TEAM:
+                return translate('workspace.type.collect');
+            default:
+                throw new Error(`Don't know a friendly workspace name for this workspace type`);
+        }
+    }, [workspaceType, translate]);
+
+    if (isWide === undefined) {
+        return <></>;
+    }
+
+    return (
+        <View style={[isWide ? styles.flexRow : styles.flexColumn, isWide && styles.gap5, styles.highlightBG, styles.br3, styles.pv5, styles.pl5]}>
+            <View style={[styles.flexRow, isWide && styles.flex1, styles.gap3, !isWide && [styles.mb3, styles.mr2], styles.alignItemsCenter]}>
+                <Avatar
+                    imageStyles={[styles.alignSelfCenter]}
+                    size={CONST.AVATAR_SIZE.DEFAULT}
+                    source={workspaceIcon}
+                    fallbackIcon={fallbackWorkspaceIcon}
+                    name={title}
+                    type={CONST.ICON_TYPE_WORKSPACE}
+                />
+                <Text
+                    numberOfLines={1}
+                    style={[styles.flexGrow1, styles.textStrong]}
+                >
+                    {title}
+                </Text>
+                {!isWide && (
+                    <ThreeDotsMenu
+                        menuItems={[]}
+                        anchorPosition={{top: 0, right: 0}}
+                        iconTooltip={undefined}
+                        icon={undefined}
+                        iconFill={undefined}
+                        iconStyles={undefined}
+                        onIconPress={undefined}
+                        anchorAlignment={undefined}
+                        shouldOverlay={undefined}
+                        shouldSetModalVisibility={undefined}
+                        disabled={undefined}
```

Comment:
why do we need to explicitly pass `undefined` for these props?

Reply:
This was auto generated as ThreeDotsMenu is a JS file so .tsx didn't know which of those are required. I think I can delete now and it will still work. Testing

Reply:
please close this conversation when you confirm that the code works 🙇